mqtt/mqtt-broker/src/connection/packet.rs

```diff
@@ -105,8 +105,8 @@ impl IncomingPacketProcessor for MqttIncomingPacketProcessor {
             Packet::PubAck(puback) => ClientEvent::PubAck(puback),
             Packet::PubComp(pubcomp) => ClientEvent::PubComp(pubcomp),
             Packet::Publish(publish) => {
-                let perm = self.incoming_pub_limit.clone().acquire_owned().await;
-                ClientEvent::PublishFrom(publish, Some(perm))
+                let permit = self.incoming_pub_limit.clone().acquire_owned().await.ok();
```

Comment:
```
pub async fn acquire_owned(
    self: Arc<Self>
) -> Result<OwnedSemaphorePermit, AcquireError>
[src]
[−]

Acquires a permit from the semaphore.

The semaphore must be wrapped in an Arc to call this method. If the semaphore has been closed, this returns an AcquireError. Otherwise, this returns a OwnedSemaphorePermit representing the acquired permit.
```
So... maybe?

Reply:
It can fail, but only when `Semaphore` is dropped which is kind of not what we expect: processor holds semaphore all its life time